src/libcore/iter/adapters/mod.rs

```diff
@@ -541,6 +541,21 @@ impl<I> Iterator for StepBy<I> where I: Iterator {
             self.iter.nth(nth - 1);
         }
     }
+
+    fn try_fold<Acc, F, R>(&mut self, mut acc: Acc, mut f: F) -> R
+    where
+        F: FnMut(Acc, Self::Item) -> R,
+        R: Try<Ok = Acc>,
+    {
+        if self.first_take {
+            self.first_take = false;
+            match self.iter.next() {
+                None => return Try::from_ok(acc),
+                Some(x) => acc = f(acc, x)?,
+            }
+        }
+        from_fn(|| self.iter.nth(self.step)).try_fold(acc, f)
```

Comment:
Can you just use a `while let Some(x) = self.iter.nth(self.step)` loop here instead?  It'll do the same thing, but without introducing a new closure that would regress https://github.com/rust-lang/rust/pull/62429.

Reply:
Thanks, I forgot about that. I turned the closures into functions instead, that should have the same benefits, correct? I figured this will make it easier to potentially undo these changes in the future.